tests/unittests/core/io/test_experiment_builder.py

```diff
@@ -249,11 +252,17 @@ def test_get_from_args_no_hit(config_file):
 
 
 @pytest.mark.usefixtures("with_user_tsirif")
-def test_get_from_args_hit(config_file, random_dt, new_config):
+def test_get_from_args_hit(monkeypatch, raw_config, random_dt, new_config):
     """Try building experiment view when in db"""
-    cmdargs = {"name": "supernaekei", "config": config_file}
+    cmdargs = {"name": "supernaekei", "config": raw_config}
+
+    with OrionState(experiments=[new_config], trials=[]) as cfg:
+
+        def get_storage(*args, **kwargs):
+            return cfg.storage
+
+        monkeypatch.setattr(experiment_builder, "setup_storage", get_storage)
```

Comment:
Why do you need to replace `setup_storage` in the next couple tests? Isn't the global `storage` config set with `OrionState`?

Reply:
because of `get_from_args` which will call `setup_storage` creating its own instance of the storage with the config coming form the `get_from_args` not the global config

Reply:
But it will use the global config if nothing is defined in a local config file passed to the arguments.

Reply:
I added comments for clarity so people dont lose time on that :)

Reply:
and `get_from_args` gets a different storage because it is configured to do so from ``orion_config.yaml``

Reply:
Ah I see, previous version of the tests were not using the local storage config as well, since the singleton was already instantiated by OrionState.

Perhaps using a config file without a config for the storage would be simpler and less confusing than a mock.

Reply:
done